instrumentation/opentelemetry-instrumentation-grpc/tests/test_filters.py

```diff
@@ -0,0 +1,357 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import collections
+
+import grpc
+import pytest
+
+from opentelemetry.instrumentation.grpc import filters
+
+
+class _HandlerCallDetails(
+    collections.namedtuple(
+        "_HanlderCallDetails",
+        (
+            "method",
+            "invocation_metadata",
+        ),
+    ),
+    grpc.HandlerCallDetails,
+):
+    pass
+
+
+class _UnaryClientInfo(
+    collections.namedtuple("_UnaryClientInfo", ("full_method", "timeout"))
+):
+    pass
+
+
+class _StreamClientInfo(
+    collections.namedtuple(
+        "_StreamClientInfo",
+        ("full_method", "is_client_stream", "is_server_stream", "timeout"),
+    )
+):
+    pass
```

Comment:
nit I would prefer [`@dataclass`](https://docs.python.org/3/library/dataclasses.html) over named tuple

Reply:
Either `collections.namedtuple` or `@dataclass` is fine. I aligned the implementation here with the one in `grpcext`.

https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/instrumentation/opentelemetry-instrumentation-grpc/src/opentelemetry/instrumentation/grpc/grpcext/_interceptor.py#L28-L40